tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java

```diff
@@ -0,0 +1,285 @@
+        }
+
+        switch( id )
+        {
+        case IProblem.ImportNotFound:
+            ImportNotFoundProposal( context, problem, proposals );
+            break;
+        case IProblem.UndefinedType:
+        case IProblem.JavadocUndefinedType:
+            undefinedType( context, problem, proposals );
+            break;
+        default:;
+        }
+    }
+
+    private void undefinedType(
+        IInvocationContext context, IProblemLocation problem, Collection<IJavaCompletionProposal> proposals )
+    {
+        ASTNode selectedNode = problem.getCoveringNode( context.getASTRoot() );
+        String fullyQualifiedName = null;
+
+        if( selectedNode instanceof Name )
+        {
+            Name node = (Name) selectedNode;
+            fullyQualifiedName = node.getFullyQualifiedName();
+        }
+
+        List<String> serviceWrapperList;
+        List<String> servicesList;
+        boolean depWrapperCanFixed = false;
+
+        try
+        {
+            serviceWrapperList = TargetPlatformUtil.getServiceWrapperList().getServiceList();
+            servicesList = TargetPlatformUtil.getServicesList().getServiceList();
+
+            for( String wrapper : serviceWrapperList )
+            {
+                if( wrapper.endsWith( fullyQualifiedName ) )
+                {
+                    ServiceContainer bundle = TargetPlatformUtil.getServiceWrapperBundle( fullyQualifiedName );
+                    createDepProposal( context, proposals, bundle );
+                }
+            }
+
+            if( !depWrapperCanFixed )
+            {
+                for( String service : servicesList )
+                {
+                    if( service.endsWith( fullyQualifiedName ) )
+                    {
+                        ServiceContainer bundle = TargetPlatformUtil.getServiceBundle( service );
+                        createDepProposal( context, proposals, bundle );
+                    }
+                }
+            }
+        }
+        catch( Exception e )
+        {
+            GradleCore.logError( "Gradle dependence got error", e );
+        }
+    }
+
+    private void ImportNotFoundProposal(
+        IInvocationContext context, IProblemLocation problem, Collection<IJavaCompletionProposal> proposals )
+    {
+        ASTNode selectedNode = problem.getCoveringNode( context.getASTRoot() );
+
+        if( selectedNode == null )
+        {
+            return;
+        }
+
+        ImportDeclaration importDeclaration =
+            (ImportDeclaration) ASTNodes.getParent( selectedNode, ASTNode.IMPORT_DECLARATION );
+
+        if( importDeclaration == null )
+        {
+            return;
+        }
+
+        String importName = importDeclaration.getName().toString();
+        List<String> serviceWrapperList;
+        List<String> servicesList;
+        boolean depWrapperCanFixed = false;
+
+        try
+        {
+            serviceWrapperList = TargetPlatformUtil.getServiceWrapperList().getServiceList();
```

Comment:
the serviceList should be a "Set" not a "list" then we can say "set.contains(...)"